capsules/src/print_log.rs

```diff
@@ -0,0 +1,276 @@
+//! Provides userspace with access to a serial interface whose output
```

Comment:
The key property is that the write is synchronous: this is what allows it to be in-order. But it bounds the length of what you can write. I avoided calling it a Console because you can't read from it. But I agree that a different name is needed. I thought log was a good name because of the lack of reading?

Reply:
I personally don't associate all consoles as being read+write; for example the console object in javascript is write only AFAIK (https://developer.mozilla.org/en-US/docs/Web/API/console). That said, I think `InOrderLog` is fine; I just don't like the collision with the log capsule meant for logging to flash.

Reply:
We could change Console to ReadWriteConsole and this to InOrderConsoleLog.... :)

Reply:
I suppose any naming should wait for the conclusions of discussing this on the core call -- if we wanted these to implement the same device we would need to port over the read functionality from Console anyway